test/test_pipeline/implementations/test_SparseOneHotEncoder.py

```diff
@@ -0,0 +1,112 @@
+
+
+class TestSparseOneHotEncoder(unittest.TestCase):
+    def test_sparse1(self):
+        self._fit_then_transform(sparse1_1h.todense(), sparse1)
+
+    def test_sparse2(self):
+        self._fit_then_transform(sparse2_1h.todense(), sparse2)
+
+    def test_sparse2_csr(self):
+        self._fit_then_transform(sparse2_csr_1h.todense(), sparse2_csr)
+
+    def _fit_then_transform(self, expected, input):
+        # Test fit_transform
+        input_copy = input.copy()
+        ohe = SparseOneHotEncoder()
+        transformation = ohe.fit_transform(input)
+        self.assertIsInstance(transformation, scipy.sparse.csr_matrix)
+        assert_array_almost_equal(expected.astype(float),
+                                  transformation.todense())
+        self._check_arrays_equal(input, input_copy)
+
+        # Test fit, and afterwards transform
+        ohe2 = SparseOneHotEncoder()
+        ohe2.fit(input)
+        transformation = ohe2.transform(input)
+        self.assertIsInstance(transformation, scipy.sparse.csr_matrix)
+        assert_array_almost_equal(expected, transformation.todense())
+        self._check_arrays_equal(input, input_copy)
+
+    def _check_arrays_equal(self, a1, a2):
+        if scipy.sparse.issparse(a1):
+            a1 = a1.toarray()
+        if scipy.sparse.issparse(a2):
+            a2 = a2.toarray()
+        assert_array_almost_equal(a1, a2)
+
+    def test_transform_with_unknown_value(self):
+        input = np.array(((0, 1, 2, 3, 4, 5), (0, 1, 2, 3, 4, 5))).transpose()
+        ips = scipy.sparse.csr_matrix(input)
+        ohe = SparseOneHotEncoder()
+        ohe.fit(ips)
+        test_data = np.array(((0, 1, 2, 6), (0, 1, 6, 7))).transpose()
+        tds = scipy.sparse.csr_matrix(test_data)
+        output = ohe.transform(tds).todense()
+        self.assertEqual(3, np.sum(output))
+
+    def test_classification_workflow(self):
+        task = openml.tasks.get_task(254)
+        X, y = task.get_X_and_y()
+        
+        imputation = sklearn.impute.SimpleImputer(strategy='constant', fill_value=2)
+        ohe = SparseOneHotEncoder()
+        tree = sklearn.tree.DecisionTreeClassifier(random_state=1)
+        
+        pipeline = sklearn.pipeline.Pipeline((
+            ('imput', imputation),
+            ('ohe', ohe),
+            ('tree', tree)))
+
+        X = CategoryShift().transform(X)
+        X_train, X_test, y_train, y_test = \
+            sklearn.model_selection.train_test_split(X, y, random_state=3,
+                                                     train_size=0.5,
+                                                     test_size=0.5)
+        
+        X_train = scipy.sparse.csc_matrix(X_train)
+        X_test = scipy.sparse.csc_matrix(X_test)
+        
+        pipeline.fit(X_train, y_train)
+        self.assertEqual(np.mean(y_train == pipeline.predict(X_train)), 1)
+        # With an incorrect copy operation the OneHotEncoder would rearrange
```

Comment:
Would there be better ways of checking that the workflow actually behaves as expected?

Reply:
Fixed. This code were from the original auto-sklearn OneHotEnconder test.